Updated bdk-cli/installation content to v0.2.1 (#46)
bitcoindevkit/bitcoindevkit.org: content/bdk-cli/installation.md

`cargo version`, which should print something like:

```

> cargo 1.52.0 (69767412a 2021-04-21)

Reviewer: Sorry this is such a moving target. The latest `stable` rust since June 17 2021 is 1.53.0 
```suggestion
cargo 1.53.0 (4369396ce 2021-04-27)
```

```

If you really don't want to pipe the output of `curl` into `sh`, you can also try using a
[Docker image](https://hub.docker.com/_/rust) and working inside of it, but that's meant for more
advanced users and won't be covered in this guide.

{{% notice note %}}

> At the time of writing, the project requires cargo >= 1.52.0, which is the latest stable as of

Reviewer: ```suggestion
At the time of writing, the project requires cargo >= 1.46.0, which is our minimum supported rust version (MSRV) as of
```

the GitHub repository, by running:

```bash

> cargo install --git https://github.com/bitcoindevkit/bdk-cli --features="compiler"

Reviewer: Sorry but I think this line should also include `esplora` since it's mentioned on the "Interface" page. Also the quotes aren't needed.
```suggestion
cargo install --git https://github.com/bitcoindevkit/bdk-cli --features=esplora,compiler
```

Author: :P Not a problem at all!

> cargo install --git https://github.com/bitcoindevkit/bdk-cli --features=esplora,compiler,electrum,repl

Reviewer: You only need to add `compiler` to this list, `electrum` and `repl` are already included as part of the "default" features in the `bdk-cli` Cargo.toml.

Author: How about esplora? 
It affects the result in the interface.md where we display the available OPTIONS for `bdk-cli wallet`.

Also wondering if we should add `compact_filters`. My reasoning here is to show all available feature options and since it will be added soon, we could just include it here although it would temporarily break.

Author: Also, what is `compiler` used for in bdk-cli?

I've used it when installing bdk to get the `minisciptc` example binary. Which although a separate issue, I think would be nice to have as part of bdk-cli. The flow to use subcommands would then be `key -> compile -> wallet`

Reviewer: Good point. For all options then you should include esplora and sure I think cbf options are stable now so can add compact_filters too but then we should wait to merge this PR until that PR is in.

Author: Never mind the comment on `compiler`, i think i have multiple versions of bdk-cli and i wasnt seeing the `compile` subcommand with `bdk-cli help`. So that flow already exists :P

Author: So then, there can be an update to the compiler content as well to use `bdk-cli compile` rather than `miniscriptc`, since it is under the bdk-cli subgroup.

Reviewer: It's true the `bdk-cli compile` sub-command will compile a policy to a descriptor, and the `bdk-cli wallet policies` sub-command will display the spending polices which I believe covers everything that the `miniscriptc` example is doing.

# most commonly used features
```

This command may take a while to finish, since it will fetch and compile all the
dependencies and the `bdk` library itself. Once it's done, you can check if everything went fine by
running `bdk-cli --help` which should print something like this:

```text
BDK CLI 0.2.1-dev
Alekos Filini <alekos.filini@gmail.com>:Riccardo Casatta <riccardo@casatta.it>:Steve Myers <steve@notmandatory.org>
Top level options and command modes

USAGE:
    bdk-cli [OPTIONS] <SUBCOMMAND>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

OPTIONS:
    -n, --network <NETWORK>    Sets the network [default: testnet]

SUBCOMMANDS:
    help      Prints this message or the help of the given subcommand(s)
    key       Key management sub-commands
    repl      Enter REPL command loop mode
    wallet    Wallet options and sub-commands
```

An example command to sync a testnet wallet looks like this:

```bash

> bdk-cli wallet -w bdk0rigins --descriptor "wpkh(tprv8ZgxMBicQKsPexGYyaFwnAsCXCjmz2FaTm6LtesyyihjbQE3gRMfXqQBXKM43DvC1UgRVv1qom1qFxNMSqVAs88qx9PhgFnfGVUdiiDf6j4/0/*)" --server ssl://electrum.blockstream.info:60002 sync 

Reviewer: a small nit, but I think it's clearer to name the wallet "example" , so `-w example` to make it clear there is no special meaning to the wallet name.

the GitHub repository, by running:

```bash
# all features

> cargo install --git https://github.com/bitcoindevkit/bdk-cli --features="compact_filters,esplora"

Reviewer: I think you should take our `compact_filters` for now since it requires more setting to make it work (the below example command would fail), and add the `compiler` optional feature 
```suggestion
cargo install --git https://github.com/bitcoindevkit/bdk-cli --features="esplora,compiler"
```
